pkg/controllers/management/drivers/kontainerdriver/kontainerdriver.go

```diff
@@ -107,30 +107,51 @@ func (l *Lifecycle) driverExists(obj *v3.KontainerDriver) bool {
 
 func (l *Lifecycle) download(obj *v3.KontainerDriver) (*v3.KontainerDriver, error) {
 	driver := drivers.NewKontainerDriver(obj.Spec.BuiltIn, obj.Name, obj.Spec.URL, obj.Spec.Checksum)
-	err := driver.Stage()
+
+	if !driver.Exists() {
```

Comment:
You need to check the condition is already unknown or not. It don't need to be updated if it is already `unknown`

Reply:
I don't think we should check if the condition already unknown, it is redundant and not necessary because there is not any effect.